NbMGui/src/main/java/nbmgui/Controller.java

```diff
@@ -139,32 +138,33 @@ private void fillProgramDataView(List<String> programDataList) {
         for (String lineStr : programDataList) {
             CheckBox line = new CheckBox(lineStr);
             line.setPadding(new Insets(1));
+            line.setOnAction(new EventHandler<ActionEvent>() {
+                @Override
+                public void handle(ActionEvent event) {
+                    if (event.getSource() instanceof CheckBox) {
+                        CheckBox breakpoint = (CheckBox) event.getSource();
+                        if (breakpoint.isSelected())
+                            machine.addBreakpoint(Integer.valueOf(breakpoint.getText().substring(2, 5)));
+                        else
+                            machine.removeBreakpoint(Integer.valueOf(breakpoint.getText().substring(2, 5)));
+                    }
+                }
+            });
             programData.getChildren().add(line);
         }
         programDataView.setContent(programData);
     }
 
-    private void setCurrentPcOfUI() {
+    private void highlightNextInstructionToBeExecuted() {
```

Comment:
I am still not very happy with this method. Going thru all the lines, touching each line with either highlighting or low low-lighting it … Wouldn't it be much more straight forward to have all information of a program line (address, reference to the Checkbox holding the line) in a map where we can access it without going thru the whole program?